xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -343,6 +345,13 @@ COverlay* CRenderer::Convert(CDVDOverlaySSA* o, double pts)
   int targetWidth = m_rv.Width();
   int targetHeight = m_rv.Height();
   int useMargin;
+  int videomode = CServiceBroker::GetGUI()->GetStereoscopicsManager().GetStereoModeOfPlayingVideo();
+  // Render subtitle of half-sbs and half-ou video in full screen, not in half screen
+  // know issue: can't identify sbs and ou video that don't need to change source size
+  if (videomode == RENDER_STEREO_MODE_SPLIT_VERTICAL)
+    sourceWidth = m_rs.Width() * 2;
+  if (videomode == RENDER_STEREO_MODE_SPLIT_HORIZONTAL)
```

Comment:
```suggestion
  else if (videomode == RENDER_STEREO_MODE_SPLIT_HORIZONTAL)
```

videomode can be `RENDER_STEREO_MODE_SPLIT_VERTICAL`, `RENDER_STEREO_MODE_SPLIT_HORIZONTAL` (or something else), but not both at the same time.